features/nanostack/mbed-mesh-api/source/WisunInterface.cpp

```diff
@@ -315,13 +316,29 @@ mesh_error_t WisunInterface::validate_network_regulatory_domain(uint8_t regulato
     return MESH_ERROR_NONE;
 }
 
-mesh_error_t WisunInterface::set_network_phy_mode_and_channel_plan_id(uint8_t phy_mode_id, uint8_t channel_plan_id)
+mesh_error_t WisunInterface::set_network_domain_configuration(uint8_t regulatory_domain, uint8_t phy_mode_id, uint8_t channel_plan_id)
 {
-    int status = ws_management_phy_mode_id_set(get_interface_id(), phy_mode_id);
+    int status = ws_management_domain_configuration_set(get_interface_id(), regulatory_domain, phy_mode_id, channel_plan_id);
     if (status != 0) {
         return MESH_ERROR_UNKNOWN;
     }
-    status = ws_management_channel_plan_id_set(get_interface_id(), channel_plan_id);
+
+    return MESH_ERROR_NONE;
+}
+
+mesh_error_t WisunInterface::get_network_domain_configuration(uint8_t *regulatory_domain, uint8_t *phy_mode_id, uint8_t *channel_plan_id)
+{
+    int status = ws_management_domain_configuration_get(get_interface_id(), regulatory_domain, phy_mode_id, channel_plan_id);
+    if (status != 0) {
+        return MESH_ERROR_UNKNOWN;
+    }
+
+    return MESH_ERROR_NONE;
+}
+
+mesh_error_t WisunInterface::validate_network_domain_configuration(uint8_t regulatory_domain, uint8_t phy_mode_id, uint8_t channel_plan_id)
+{
+    int status = ws_management_domain_configuration_validate(get_interface_id(), regulatory_domain, phy_mode_id, channel_plan_id);
     if (status != 0) {
         return MESH_ERROR_UNKNOWN;
```

Comment:
we not have a better error than unknot available?

Reply:
Actually we have these error types in mesh API:
MESH_ERROR_NONE
MESH_ERROR_UNKNOWN
MESH_ERROR_MEMORY
MESH_ERROR_STATE
MESH_ERROR_PARAM

But Nanostack API documentation (Doxygen) doesn't tell the reason for error. It only says: "return <0 Fail" for every WS management function.

We should probably first check that error values are correctly set and documented in Nanostack API.

Reply:
MESH_ERROR_PARAM may be good select for indicate proper information